### PR TITLE
9 api delete comment by comment

### PR DESCRIPTION
**Pull request by JenWarman**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -271,17 +271,22 @@ describe('nc-news API', () => {
...
-            .send({ inc_votes: 1})
-            .expect(200)
-            .then(({body}) => {
-                body.votes.forEach((vote) => {
-                    expect(vote).toHaveProperty("votes", 101)
+                .delete('/api/comments/1')
+                .expect(204)
+                .then(({ body }) => {
+                    expect(body).toEqual({});
+                });
```
Comment (nc-mohammed-dastgir) [inline __tests__/app.test.js:281]: We dont need to have an expect here for body. The status code 204 does not allow you to send anything back so testing this is testing how express works under the hood and it will always pass so this can and should be removed, which means you wont even need a .then here either

File: __tests__/app.test.js [hunk 2]
```diff
@@ -271,17 +271,22 @@ describe('nc-news API', () => {
...
+        })
+        test('400: responds with 400 error message if comment_id is invalid', () => {
+            return request(app)
+                .delete('/api/comments/iamnotacommentid')
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toBe('Bad request')
```
Comment (nc-mohammed-dastgir) [inline __tests__/app.test.js:288]: Nice, but dont forget to also test for a valid but none existent comment_id. E.g /api/comments/9999

File: controllers/topics.controllers.js [hunk 3]
```diff
@@ -66,13 +66,24 @@ exports.postNewComment = (request, response, next) => {
...
+
+exports.deleteCommentById = (request, response, next) => {
+    const { comment_id } = request.params;
+    deleteComment(comment_id)
+        .then(() => {
+            response.status(204).send();
+        })
+        .catch((err) => {
+            next(err);
+        })
```
Comment (nc-mohammed-dastgir) [inline controllers/topics.controllers.js:88]: Nice, but a really nice alternative to the response is using `res.sendStatus(204)` which also doesnt need an empty trailing .send()

File: models/topics.models.js [hunk 4]
```diff
@@ -61,23 +61,27 @@ exports.fetchComments = (article_id) => {
...
+        .then(({ rows }) => {
+            if (rows.length === 0) {
+                return Promise.reject({ status: 404, msg: 'Request not found' })
+            }
+            return rows;
+        })
+}
+
+exports.deleteComment = (comment_id) => {
+    return db.query(`DELETE FROM comments WHERE comment_id = $1`, [comment_id])
```
Comment (nc-mohammed-dastgir) [inline models/topics.models.js:86]: Nice

**Review by nc-mohammed-dastgir — COMMENTED** [review]
Nice work. I have left some comments for you to read below. Once you have made the requested changes, you can merge in directly without needing to send us an updated pr
